From: Rob Evers <revers@redhat.com>
Date: Mon, 4 Jan 2010 22:51:34 -0500
Subject: [scsi] lpfc: Made TigerShark set up and use single FCP EQ
Message-id: <20100104225141.24386.11122.sendpatchset@localhost.localdomain>
Patchwork-id: 22295
O-Subject: [RHEL5.5 PATCH 3/18] Made TigerShark set up and use single FCP EQ
	only under INTx interrupt mode (CR: 96645)
Bugzilla: 549763
RH-Acked-by: Tomas Henzl <thenzl@redhat.com>
RH-Acked-by: Mike Christie <mchristi@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=549763

Made TigerShark set up and use single FCP EQ only under INTx interrupt mode (CR: 96645)

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/lpfc/lpfc_init.c b/drivers/scsi/lpfc/lpfc_init.c
index a2f46b8..fb8d66e 100644
--- a/drivers/scsi/lpfc/lpfc_init.c
+++ b/drivers/scsi/lpfc/lpfc_init.c
@@ -7640,6 +7640,9 @@ lpfc_pci_probe_one_s4(struct pci_dev *pdev, const struct pci_device_id *pid)
 			error = -ENODEV;
 			goto out_free_sysfs_attr;
 		}
+		/* Default to single FCP EQ for non-MSI-X */
+		if (phba->intr_type != MSIX)
+			phba->cfg_fcp_eq_count = 1;
 		/* Set up SLI-4 HBA */
 		if (lpfc_sli4_hba_setup(phba)) {
 			lpfc_printf_log(phba, KERN_ERR, LOG_INIT,
